lib/modules/apostrophe-images-widgets/views/widgetBase.html

```diff
@@ -1,49 +1,55 @@
-          />
-          {%- block credit -%}
-            {%- if image.credit -%}
-              {%- if image.creditUrl -%}
-                <a href="{{ image.creditUrl }}">
+      {%- if data.widget._pieces.length > 0 -%}
+        {%- for entry in data.widget._pieces -%}
+          {# Works whether there's a relationship in the join or not. Normally there always #}
+          {# is for slideshows, but just in case someone really hates cropping... -Tom #}
+          {%- set image = entry.item or entry -%}
+          {%- set relationship = entry.relationship -%}
+          <div data-slideshow-item
+            class="apos-slideshow-item{% if loop.first %} apos-current{% endif %}{% block itemClass %}{% endblock %} apos-slideshow-item--{{ image.attachment.extension }}"
+            style="background-image:
+              url({{ apos.attachments.url(image.attachment, { size: data.options.size, crop: relationship }) }})
+              {%- if apos.attachments.hasFocalPoint(relationship) -%}
+                ; background-position: {{ apos.attachments.focalPointToBackgroundPosition(relationship) }}
               {%- endif -%}
-              {{ image.credit }}
-              {%- if image.creditUrl -%}
-                </a>
+              {%- if data.options.aspectRatio and (image.attachment.extension == 'svg') -%}
+                ; width: 100%; padding-bottom: {{ 100 * data.options.aspectRatio[1] / data.options.aspectRatio[0] }}%; background-size: contain
               {%- endif -%}
-            {%- endif -%}
-          {%- endblock -%}
-          {%- block description -%}
-            {%- if image.description -%}
-              <div>{{ image.description | nlbr }}</div>
-            {%- endif -%}
-          {%- endblock -%}
+            ">
+            {%- block title -%}
+              <h4 class="apos-image-widget-image-title">
+                {{ image.title }}
+              </h4>
+            {%- endblock -%}
+            <img
+              alt="{% block imgAlt %}{{ apos.images.getImageAlt(image) }}{% endblock %}"
+              data-image src="{% block imgSrc %}{{ apos.attachments.url(image.attachment, { size: data.options.size, crop: relationship }) }}{% endblock %}"
+            />
+            {%- block credit -%}
+              {%- if image.credit -%}
+                {%- if image.creditUrl -%}
+                  <a href="{{ image.creditUrl }}">
+                {%- endif -%}
+                {{ image.credit }}
+                {%- if image.creditUrl -%}
+                  </a>
+                {%- endif -%}
+              {%- endif -%}
+            {%- endblock -%}
+            {%- block description -%}
+              {%- if image.description -%}
+                <div>{{ image.description | nlbr }}</div>
+              {%- endif -%}
+            {%- endblock -%}
+          </div>
+        {%- endfor -%}
+      {%- elseif data.widget._edit -%}
+        <div class="apos-ui apos-empty-image">
+          <p>No images are currently selected.</p>
```

Comment:
Oh, this will need to be in a localization helper:

```
<p>{{ __ns('apostrophe', 'No images are currently selected.') }}</p>
```

I forgot about that.